### PR TITLE
KBI0024: Create and update a 7-Zip archive in a RIA store

### LINKED ISSUE #47: What is the preferred strategy for creating and updating an archive.7z in a RIA store?
Origin: [incf NeuroStars (Software support, datalad)](https://neurostars.org/t/what-is-the-preferred-strategy-for-creating-and-updating-an-archive-7z-in-a-ria-store/25683) on 2023-04-26

TODO (not necessarily to be performed in this order)
- [x] Inform OP/Add reference to this issue at origin
- [x] Clarifying Qs asked or not needed
- [x] Nature of the issue is understood
- [x] Inform OP about resolution

### PR DESCRIPTION
**Pull request by loj**
A write-up demonstrating how to create/update a RIA store 7-Zip archive using a script created by @mih.

Closes #47

### CODE REVIEW

File: kbi/0024/index.rst [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+:authors: Laura Waite <laura@waite.eu>, Michael Hanke <michael.hanke@gmail.com>
+:discussion: <link>
+:keywords: RIA, archive, 7-Zip, inodes
+:software-versions: datalad_0.18.3
+
+When working with `RIA stores`_, it is possible to compress the
+``annex/objects`` into `7-zip`_ archives. With this approach, the entire
+annex object store can be put into an archive and remain fully accessible while
+minimizing inodes, regardless of file number and size. This is beneficial
```
Comment (mslw) [inline kbi/0024/index.rst:16]:
```suggestion
minimizing `inodes`_, regardless of file number and size. This is beneficial
```

Maybe also add a link to the term definition - to me this used to be a foreign sounding word. Not necessary though, we use many other technical terms across the KBI, and it's a quick search away.

If this suggestion is accepted, a link should go at the bottom of the page.

File: kbi/0024/index.rst [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+for compression gains or when operating on HPC-systems with inode limitations.
+
+This document describes how to create a 7-Zip archive for an existing RIA
+store using the script shown below. The script does the following:
+
+1. Remove unnecessary files and directories within the RIA store (e.g. hooks,
+   etc). This aspect has nothing to do with creating the archive, but is useful
+   for reducing the number of inodes.
```
Comment (mslw) [inline kbi/0024/index.rst:24]:
```suggestion
1. Remove non-essential files and directories within the RIA store (e.g. hooks,
   etc). This aspect has nothing to do with creating the archive, but is useful
   for reducing the number of inodes.
```

To me _unnecessary_ begs the question why they are there in the first place, and non-essential sounds better. But maybe it's just me - in any case, this suggestion is non-essential :)

File: kbi/0024/index.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+
+
+.. literalinclude:: cleanup.sh
+   :language: bash
+   :linenos:
+   :caption: cleanup.sh
+
+.. _ria stores: http://handbook.datalad.org/en/latest/beyond_basics/101-147-riastores.html
+.. _7-Zip: https://www.7-zip.org/
```
Comment (mslw) [inline kbi/0024/index.rst:263]:
```suggestion
.. _7-Zip: https://www.7-zip.org/
.. _inodes: https://en.wikipedia.org/wiki/Inode
```

Related to my first suggestion. I'm fine with rejecting or using a different target.

**Review by mslw — APPROVED** [review]
This reads great and I completed the walk-through with no issues. I think it's especially valuable, since I could not find an explanation of creating the archive (other than that in can be created), neither in `create_sibling_ria` help message nor in the Handbook.

I left two suggestions (technically three, but two are related), but these are minor and entirely up for decision - the KBI is good to go as it is.

File: kbi/0024/index.rst (hunk 1 above)
Comment (loj) [reply kbi/0024/index.rst:16]: Good idea.

File: kbi/0024/index.rst [hunk 4]
```diff
@@ -0,0 +1,264 @@
+.. index::
+   single: RIA; archive
+.. highlight:: console
+
+KBI0024: Create and update a 7-Zip archive in a RIA store
+=========================================================
+
+:authors: Laura Waite <laura@waite.eu>, Michael Hanke <michael.hanke@gmail.com>
+:discussion: <link>
```
Comment (adswa) [inline kbi/0024/index.rst:9]:
```suggestion
:discussion: https://github.com/psychoinformatics-de/knowledge-base/issues/47
```

**Review by adswa — APPROVED** [review]
this reads great, thanks much! I will also link it in the handbook

**Comment by adswa** [comment]
linked in the handbook in https://github.com/datalad-handbook/book/pull/968

**Pull request merged**